pkg/config/example_test.go

```diff
@@ -0,0 +1,265 @@
+						Addresses: []string{"127.0.0.1:7050"},
+					}),
+					ModPolicy: config.AdminsPolicyKey,
+				},
+			},
+			Policies: map[string]*cb.ConfigPolicy{
+				config.AdminsPolicyKey: {
+					Policy: &cb.Policy{
+						Type: int32(cb.Policy_IMPLICIT_META),
+						Value: marshalOrPanic(&cb.ImplicitMetaPolicy{
+							SubPolicy: "MAJORITY Admins",
+							Rule:      cb.ImplicitMetaPolicy_MAJORITY,
+						}),
+					},
+					ModPolicy: config.AdminsPolicyKey,
+				},
+			},
+		},
+	}
+}
+
+// marshalOrPanic is a helper for proto marshal.
+func marshalOrPanic(pb proto.Message) []byte {
+	data, err := proto.Marshal(pb)
+	if err != nil {
+		panic(err)
+	}
+
+	return data
+}
+
+// createSigningIdentity returns a identity that can be used for signing transactions.
+// Signing identity can be retrieved from MSP configuration for each peer.
+func createSigningIdentity() *config.SigningIdentity {
+	privKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to generate private key: %v", err))
+	}
+
+	return &config.SigningIdentity{
+		Certificate: generateCert(),
+		PrivateKey:  privKey,
+		MSPID:       "ID",
+	}
+}
+
+// Generate a certificate for the SigningIdentity.
+func generateCert() *x509.Certificate {
+	serialNumberLimit := new(big.Int).Lsh(big.NewInt(1), 128)
+	serialNumber, err := rand.Int(rand.Reader, serialNumberLimit)
+
+	if err != nil {
+		log.Fatalf("Failed to generate serial number: %s", err)
+	}
+
+	return &x509.Certificate{
+		SerialNumber: serialNumber,
+		Subject: pkix.Name{
+			CommonName:   "Wile E. Coyote",
+			Organization: []string{"Acme Co"},
+		},
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().Add(365 * 24 * time.Hour),
+		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+		BasicConstraintsValid: true,
+	}
+}
+
+func Example_usage() {
+	// Retrieve the config for the channel
+	baseConfig := fetchChannelConfig()
+	// The config that will be updated
+	updatedConfig := proto.Clone(baseConfig).(*cb.Config)
+
+	// Must retrieve the current orderer configuration from block and modify
+	// the desired values
+	orderer := &config.Orderer{
+		OrdererType: config.ConsensusTypeKafka,
+		Kafka: config.Kafka{
+			Brokers: []string{"kafka0:9092", "kafka1:9092"},
+		},
+		Organizations: []*config.Organization{
+			{
+				Name: "OrdererOrg",
+				Policies: map[string]*config.Policy{
+					config.AdminsPolicyKey: {
+						Type: config.ImplicitMetaPolicyType,
+						Rule: "Majority Endorsement",
+					},
+					config.ReadersPolicyKey: { // Adding Readers policy
+						Type: config.ImplicitMetaPolicyType,
+						Rule: "Majority Endorsement",
+					},
+				},
+			},
+		},
+		BatchSize: config.BatchSize{
+			MaxMessageCount:   500, // Updating from 100
+			AbsoluteMaxBytes:  100,
+			PreferredMaxBytes: 100,
+		},
+		Addresses: []string{"127.0.0.1:7050"},
+	}
```

Comment:
There should really be a section here for actually retrieving the existing orderer values from the config, because it's somewhat convoluted to do so in the example this is probably fine for now (maybe have a helper function here though that returns the hardcoded values set in `fetchChannelConfig`) until FAB-17568 is done and the helper is available for retrieving the order configuration.

Reply:
I'll leave this as is for now and the example can be updated once FAB-17568 is completed.